Fix:

./lj_arch.h:127:5: error: 'TARGET_OS_IPHONE' is not defined, evaluates to 0 [-Werror,-Wundef-prefix=TARGET_OS_]

https://github.com/LuaJIT/LuaJIT/pull/1189
--- src/lj_arch.h.orig	2024-03-10 11:29:48.000000000 -0500
+++ src/lj_arch.h	2024-04-21 16:44:53.000000000 -0500
@@ -124,7 +124,7 @@
 #define LJ_TARGET_POSIX		(LUAJIT_OS > LUAJIT_OS_WINDOWS)
 #define LJ_TARGET_DLOPEN	LJ_TARGET_POSIX
 
-#if TARGET_OS_IPHONE
+#if defined(TARGET_OS_IPHONE) && TARGET_OS_IPHONE
 #define LJ_TARGET_IOS		1
 #else
 #define LJ_TARGET_IOS		0
